File: src/melius_commons/pokefenn/Melius.java
package melius_commons.pokefenn;

import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.Mod.Instance;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.network.NetworkMod;
import melius_commons.pokefenn.lib.Strings;
import melius_commons.pokefenn.packet.PacketHandler;
import melius_commons.pokefenn.proxy.CommonProxy;
import net.minecraft.creativetab.CreativeTabs;

/**
 * Created with IntelliJ IDEA.
 * User: Pokefenn
 * Date: 13/12/13
 * Time: 12:46
 */

@Mod(modid = Strings.modid, name = Strings.name, version = Strings.version)
@NetworkMod(channels = { Strings.modid }, clientSideRequired = true, serverSideRequired = false, packetHandler = PacketHandler.class)
public class Melius {

    @Instance(Strings.modid)
    public static Melius instance;

    @SidedProxy(clientSide = Strings.CLIENT_PROXY_CLASS, serverSide = Strings.SERVER_PROXY_CLASS)
    public static CommonProxy proxy;

    public static CreativeTabs tabsMelius = new CreativeTabMelius(CreativeTabs.getNextID(), Strings.modid);



}

File: src/melius_commons/pokefenn/Melius.scala
package melius_commons.pokefenn

import cpw.mods.fml.common.network.NetworkMod
import cpw.mods.fml.common.{FMLCommonHandler, Mod}
import cpw.mods.fml.common.event.{FMLPostInitializationEvent, FMLInitializationEvent, FMLPreInitializationEvent}
import cpw.mods.fml.common.Mod.{Instance, EventHandler}
import java.util.logging.Logger
import melius_commons.pokefenn.packet._
import cpw.mods.fml.relauncher.Side
import melius_commons.pokefenn.lib.Strings
import melius_commons.pokefenn.block.ModBlocks
import java.io.File
import melius_commons.pokefenn.configuration.ConfigurationHandler

/**
 * Created with IntelliJ IDEA.
 * User: Pokefenn
 * Date: 13/12/13
 * Time: 09:51

 */

@Mod(modid = "melius", name = "Melius", version = "0.0.0",  modLanguage = "scala")
@NetworkMod(channels = Array("melius"), clientSideRequired = true, serverSideRequired = false)
object Melius {




  @Instance("melius") var instance: Melius.type = null
  final val logger: Logger = Logger.getLogger("melius")



  @EventHandler def preInit(event: FMLPreInitializationEvent) {

    ConfigurationHandler.init(new File(event.getModConfigurationDirectory.getAbsolutePath + File.separator + Strings.modid + File.separator + Strings.name + ".cfg"))

    ModBlocks.init()


  }

  @EventHandler def init(evt: FMLInitializationEvent) {



  }


  @EventHandler def postInit(evt: FMLPostInitializationEvent) {



  }












}
